regimes/se/invoices.go

```diff
@@ -0,0 +1,99 @@
+package se
+
+import (
+	"github.com/invopop/gobl/bill"
+	"github.com/invopop/gobl/org"
+	"github.com/invopop/gobl/tax"
+	"github.com/invopop/validation"
+)
+
+// invoiceValidator adds validation checks to invoices which are relevant
+// for the region.
+type invoiceValidator struct {
+	inv *bill.Invoice
+}
+
+func validateInvoice(inv *bill.Invoice) error {
+	v := &invoiceValidator{inv: inv}
+	return v.validate()
+}
+
+func (v *invoiceValidator) validate() error {
+	inv := v.inv
+	return validation.ValidateStruct(inv,
+		validation.Field(&inv.Supplier,
+			validation.Required,
+			validation.By(v.validateCommon),
+			validation.By(v.validateSupplier),
+			validation.Skip,
+		),
+		validation.Field(&inv.Customer,
+			validation.Required,
+			validation.By(v.validateCommon),
+			validation.By(v.validateCustomer),
+			validation.Skip,
+		),
+	)
+}
+
+func (v *invoiceValidator) validateCommon(value any) error {
+	party, ok := value.(*org.Party)
+	if !ok || party == nil {
+		return nil
+	}
+
+	return validation.ValidateStruct(party,
+		validation.Field(&party.Name,
+			validation.Required,
+		),
+		validation.Field(&party.Addresses,
+			validation.Required,
+		),
+	)
+}
+
+// validateSupplier checks the supplier's tax ID and organization number requirements.
+// The supplier's VAT number and ID Number are always required.
+func (v *invoiceValidator) validateSupplier(value any) error {
+	party, ok := value.(*org.Party)
+	if !ok || party == nil {
+		return nil
+	}
+
+	return validation.ValidateStruct(party,
+		// Swedish Tax ID (VAT ID) is always required for the supplier,
+		// and must have the correct format.
+		validation.Field(&party.TaxID,
+			validation.Required,
+			validation.By(func(value interface{}) error {
+				tID, ok := value.(*tax.Identity)
+				if !ok || tID == nil {
+					return nil
+				}
+				return validateTaxIdentity(tID)
+			}),
+		),
+		validation.Field(&party.Identities,
+			validation.Required,
+			org.RequireIdentityKey(ValidOrgIdentityKeys...),
```

Comment:
Is this true? Usually only the VAT ID is a requirement, everything else is optional.

Reply:
This is another one of the doubts I added to `decisions.md` as I'm not fully clear on differences between `tax.Identity` vs `org.Identity` and which cases require which. You are right that we only need the VAT ID. However, the actual validation for whether that code is valid is really done in the `org.Identity` where I check the format, numbers and Luhn algorithm all check out.

It sounds to me like I should move those here and remove the `org` identities?

Reply:
Org Identities can be really useful to have with validation in case a user whats to add them to an invoice, but they are secondary to the Tax Identity which is the main code used for international transactions. Most countries, with the notable exception of the US have a Tax ID.